pkg/cli/mirror/initcmd/initcmd.go

```diff
@@ -0,0 +1,202 @@
+package initcmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/blang/semver/v4"
+	"github.com/openshift/oc-mirror/pkg/cli/mirror/list"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/oc-mirror/pkg/api/v1alpha2"
+	"github.com/openshift/oc-mirror/pkg/cli"
+	"github.com/openshift/oc-mirror/pkg/version"
+)
+
+type InitOptions struct {
+	*cli.RootOptions
+	Output string // TODO rename: format? Be consistent with `version`
+}
+
+func NewInitCommand(f kcmdutil.Factory, ro *cli.RootOptions) *cobra.Command {
+	o := InitOptions{
+		RootOptions: ro,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Output initial config template",
+		Example: templates.Examples(`
+			# Get oc-mirror initial config template
+			oc-mirror init
+			
+			# Save oc-mirror initial config to a file
+			oc-mirror init >imageset-config.yaml
+		`),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	fs := cmd.Flags()
+	fs.StringVar(&o.Output, "output", o.Output, "One of 'yaml' or 'json'.")
+	o.BindFlags(cmd.PersistentFlags())
+
+	return cmd
+}
+
+func (o *InitOptions) Validate() error {
+	if o.Output != "" && o.Output != "yaml" && o.Output != "json" {
+		return errors.New(`--output must be 'yaml' or 'json'`)
+	}
+	return nil
+}
+
+func (o *InitOptions) Run() error {
+	var err error
+
+	gitVersion := version.Get().GitVersion
```

Comment:
Using the `gitVersion` works when this is compiled by ART, but if compiled locally this will produce an unexpected result. Maybe there is a way we can set the gitVersion per the latest branch version for locally compiled binaries? You could also borrow from the `list releases` command here https://github.com/openshift/oc-mirror/blob/3c2903f578c0757bc2fc62085755ab860a3978d7/pkg/cli/mirror/list/releases.go#L189.

Reply:
Thanks, I added a note in the README about unexpected versions for local builds, which hopefully addresses your concern while avoiding a network request